src/main/java/com/weirddev/testme/intellij/builder/MethodFactory.java

```diff
@@ -169,6 +171,25 @@ private static List<Param> extractMethodParams(PsiMethod psiMethod, boolean shou
         return params;
     }
 
+
+    //analyze the exception types of the method
+    private static String extractMethodExceptionTypes(PsiMethod psiMethod ,Boolean throwSpecificExceptionTypes,boolean testable) {
+        if(!testable){
+            return null;
+        }
+        String throwsExceptions = null;
+        if(throwSpecificExceptionTypes){
+            PsiReferenceList throwsList = psiMethod.getThrowsList();
+            PsiClassType[] referencedTypes = throwsList.getReferencedTypes();
+            throwsExceptions = Arrays.stream(referencedTypes)
+                    .map(PsiClassType::resolve)
+                    .filter(Objects::nonNull)
+                    .map(PsiClass::getQualifiedName)
+                    .filter(Objects::nonNull)
+                    .collect(Collectors.joining(","));
+        }
+        return throwsExceptions == null ? "Exception" : throwsExceptions;
```

Comment:
I think in case non found -  `throwsException` would be empty string rather then null, if so please replace with 
`return throwsException.isEmpty() ? "Exception" : throwsException;`
would also be good to change/add integration test to throw a more specific exception (i.e. `java.io.IOException`) rather then `Exception`  in order to verify mapping logic.

Reply:
ok，Let me modify the code then submit it

Reply:
1. in case non found - throwsException would be empty string rather then null  --YES
2. in case "Declare specific test method thrown exception types " is not checked --throwsException is null
3. in cased "Declare specific test method thrown exception types " is checked and The testing method does not contain any exceptions--- then test method does not throw any exceptions.
I prefer to  case 3，
so ,you prefer to throwing Exception in case 3?

Reply:
Thank you for optimizing the code. I am modifying the code and then submit.